cmd/nomos/vet/vet.go

```diff
@@ -43,6 +45,13 @@ func init() {
 	Cmd.Flags().BoolVar(&keepOutput, "keep-output", false,
 		`If enabled, keep the hydrated output`)
 
+	Cmd.Flags().IntVar(&threshold, "threshold", system.DefaultMaxObjectCount,
+		`If greater than zero, error if any repository contains more than the specified number of objects `)
+	// Set value to zero when `--threshold` is passed without a value.
+	// This requires the flag and value to be sent in the same argument, like
+	// `--threshold=1000`, instead of `--threshold 1000`.
```

Comment:
We should allow both `--threshold=1000` and `--threshold 1000`

Reply:
That's not possible with the cobra CLI library we're using. I'm not even sure it's possible with the Go flags library it wraps.

The only way to make the value optional is to set the `NoOptDefVal` field. This works by only checking a single argument and optionally allowing an `=` delimiter.

Reply:
Personally, I think this makes for confusing UX, and suggests that having 2 different defaults is unusual and undesirable.

Reply:
There's a feature request open for the underlying pflag library: https://github.com/spf13/pflag/issues/321

One of the comments notes:

> Python's argparse module actually implements the behavior you desire, but I have yet to use another option/flag parser that has the same behavior.